qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -172,29 +187,91 @@ def assign_user_parameters(
                 ({len(self._user_parameters)}).
                 """
                 )
-            param_binds = {param: values[i] for i, param in enumerate(self._user_parameters)}
-
-        if self._user_param_binds is None:
-            self._user_param_binds = param_binds
+            values = {p: values[i] for i, p in enumerate(self._user_parameters)}
         else:
-            self._user_param_binds.update(param_binds)
+            if not isinstance(values, dict):
+                raise ValueError(
+                    f"""
+                'values' must be of type Dict or Sequence.
+                Type {type(values)} is not supported.
+                """
```

Comment:
I suggest indenting these three lines one additional level, to match the preceding line.